### PR TITLE
Replace latin-1 characters so that gene names are consistent

### PR DESCRIPTION
**Pull request by achave11-ucsc**

### CODE REVIEW

**Comment by dsotirho-ucsc** [comment]
Ticket https://github.com/DailyDreaming/load-project/issues/114

File: convert_matrices.py [hunk 1]
```diff
@@ -968,13 +970,19 @@ class GSE127969(Converter):
...
+                'GSE127969_counts_TPM_ALL.csv.gz',
+                sep='\t',
+                row_filter=self._filter,
+                kwargs=dict(encoding='latin-1'))
         )
 
+    def _filter(self, row: List[str]):
+        if row[0].startswith('\x93') or row[0].endswith('\x94'):
+            gene = row[0].replace('\x93', '')
+            row[0] = gene.replace('\x94', '')
```
Comment (dsotirho-ucsc) [inline convert_matrices.py]:
```suggestion
            row[0] = row[0].replace('\x93', '').replace('\x94', '')
```

**Review by dsotirho-ucsc — APPROVED** [review]
`.replace()` can be chained to avoid creating temp variable `gene`.
Other than that it looks good to me.

File: convert_matrices.py [hunk 2]
```diff
@@ -968,13 +970,18 @@ class GSE127969(Converter):
...
-            CSV('GSE127969_counts_TPM_ALL.csv.gz', sep='\t')
+            CSV(
+                'GSE127969_counts_TPM_ALL.csv.gz',
+                sep='\t',
+                row_filter=self._filter,
+                kwargs=dict(encoding='latin-1'))
         )
 
+    def _filter(self, row: List[str]):
+        if row[0].startswith('\x93') or row[0].endswith('\x94'):
```
Comment (hannes-ucsc) [inline convert_matrices.py]: Are you sure that the only occurrences of these characters is at the beginning or end of a cell?

Are you sure that they only occur in the first column?

If the answer to either to both of these is yes, start thinking about how you can optimize statements: For example, if you know a character occurs at the beginning of a string, do you really need to scan the entire string? Same with the end. If the argument to startswith() is a single character string, can you optimize that expression?

If then answer to either of these question is no, we need to rethink this approach.

File: convert_matrices.py [hunk 3]
```diff
@@ -56,13 +56,15 @@ class CSV:
...
+    kwargs: Optional[dict] = None
 
     def to_mtx(self, input_dir: Path, output_dir: Path):
         converter = CSVConverter(
             input_file=input_dir / self.name,
             delimiter=self.sep,
             rows_are_genes=self.rows_are_genes,
-            row_filter=self.row_filter)
+            row_filter=self.row_filter,
+            **self.kwargs)
```
Comment (hannes-ucsc) [inline convert_matrices.py]: Wouldn't this fail if kwargs is the default value of None? Did you test this for any projects other than the one with the latin1-encoded file?

Comment (hannes-ucsc) [reply convert_matrices.py]: Anyhow, I'm not a fan of the wholesale kwargs pass-though. Please just add an `encoding` field instead.

**Review by hannes-ucsc — CHANGES_REQUESTED** [review]

**Review by hannes-ucsc — APPROVED** [review]

**Comment by hannes-ucsc** [comment]
I'll run `make matrices` on  GSE127969 shortly.

**Pull request merged**